### PR TITLE
Re-doing the start of the radar color fresh

### PR DESCRIPTION
**Pull request by Zitchas**
Re-PR'ing to get rid of some weird update commits that snuck into the branch before they were on master...

Feature: This PR implements the feature request detailed and discussed in issue #{{insert number}}

This PR is for testing and figuring out what I'm doing and asking for help.

Here's what I've got so far:

- interfaces.txt has a section that specifies colors. Currently, the "special" color is the one that dictates the star color. I've changed it here for identification purposes. I have added a "radar star" line with another color. (it isn't actually used)
- radar.cpp has a list of constants that appears to be assigning each type of radar object a number. Or perhaps objects already have a number and what they are is being identified. I have added Radar::STAR = 9 to the list.
- radar.cpp also has a const Color section that I think is pulling the colors from gamedata, which I think means the interfaces.txt file. I have added a line for "radar star" to the list.
- radar.cpp has a section for drawing the radar that mentions circles without an inner. This is how planets and stars are drawn, so this could be a clue as to where the object is identified. Made a comment to this effect to help find it.
- System.cpp has a section that looks like it is identifying what things are, and has a object.isStar = line, but I don't know what it does. I added a comment to it so that it would be easy to find. Another section talked about using "friend system" to identify stuff.

Thanks to RisingLeaf who had already made some suggestions and pointers on the former PR #129 ,

### CODE REVIEW

File: data/interfaces.txt [hunk 1]
```diff
@@ -101,10 +101,11 @@ color "radar friendly" .4 .6 1. 0.
 color "radar unfriendly" .8 .8 .4 0.
 color "radar hostile" .85 .3 .2 0.
 color "radar inactive" .4 .4 .4 0.
-color "radar special" 1. 1. 1. 0.
+color "radar special" 0.5 1. 0.5 0.
 color "radar anomalous" .7 0. 1. 0.
 color "radar blink" 1. 1. 1. 0.
 color "radar viewport" 0. .3 0. 0.
+color "radar star" 0.7 1. 1.
```
Comment (RisingLeaf) [inline data/interfaces.txt:108]:
```suggestion
color "radar star" 0.7 1. 1. 0.
```

**Comment by warp-core** [comment]
Radar drawing is handled in Engine::Draw() (I'm pretty sure there's a comment above the relevant part, it's a fiarly long method), with various parts of the Radar class involved. I don't think you'll ffind much in StellarObject or System about it.

**Comment by warp-core** [comment]
The parts of System about identifying what the StellarObject is are more just about the message you get when attempting to land on something you can't land on.
AI also makes use of StellarObject::IsStar() to identify planets that ships with atmosphere scan should consider.

**Comment by vitalchip** [comment]
You'll need to chnge StellarObject.

Once you haave the new radar colour (STAR?)  then you'd replece SPECIAL here.

Screenshot 2022-12-29 at 23 24 43

**Comment by Zitchas** [comment]
That's line 101? I think I got it, thanks!

**Comment by Zitchas** [comment]
Woot, not only does it built, but it works!

**Comment by Zitchas** [comment]
Thanks for your guiddance!

**Comment by vitalchip** [comment]
The 'it works!' moment is great. Even the 'hmm, it nearly works but it's a bit wonky'  is fairly decent.

**Pull request closed without merging**